src/extension-client/remix-extension.ts

```diff
@@ -117,7 +114,7 @@ export class RemixExtension<T extends Api = any> {
   /** Return the current theme when handshaked */
   public loaded(): Promise<{theme: string}> {
```

Comment:
I understand that this it not the current focus for this PR but:
seems that if `loaded()` is called after the plugin is actually loaded, the Promise is never resolved.

Reply:
for `/** Return the current theme when handshaked */` as we discussed, that is better to have a `config` plugin.

Reply:
> the Promise is never resolved.

This is not the aim of this method to be called after the plugin is loaded. It's like if you start listening on `body.onload` once the body is loaded... But we can have a variable `isLoaded` is needed.

>  that is better to have a config plugin.

Yes, I'll remove that